testing/e2e/manifests/embedded/customErrorsClient.go

```diff
@@ -0,0 +1,123 @@
+		PreferGo: true,
+		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+			d := net.Dialer{Timeout: time.Second}
+			var ns string
+			if nameserver[:len(nameserver)-1] == "." {
+				ns = nameserver[:len(ns)-1] // remove trailing period added for some reason by azure dns
+			} else {
+				ns = nameserver // no need to remove trailing period if single entry coming from k8s vnet ns server
+			}
+
+			return d.DialContext(ctx, "tcp", ns+":53")
+		},
+	}}
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		DialContext:     dialer.DialContext,
+	}}
+
+	liveServiceBody := "live service"
+	deadServiceBody := "<body>CONFIRMING CUSTOM 503 TEST MESSAGE</body>"
+	notFoundBody := "<body>CONFIRMING CUSTOM 404 TEST MESSAGE</body>"
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		// live service tests
+		liveResp, err := client.Get(os.Getenv("LIVE"))
+		if err != nil {
+			log.Printf("error sending request: %s", err)
+			w.WriteHeader(500)
+			return
+		}
+		defer liveResp.Body.Close()
+
+		body, err := ioutil.ReadAll(liveResp.Body)
+
+		if err != nil {
+			log.Printf("error reading response body: %s", err)
+			w.WriteHeader(500)
+			return
+		}
+
+		cleanedBody := cleanBody(string(body)[:(len(liveServiceBody))])
+		log.Printf("received response: %s from url %s", cleanedBody, os.Getenv("LIVE"))
+		if cleanedBody != liveServiceBody {
+			log.Printf("unexpected response body: %s, expected: %s", cleanedBody, liveServiceBody)
+			w.WriteHeader(500)
+			return
+		}
+
+		// dead service tests - 503 error codes
+		deadResp, err := client.Get(os.Getenv("DEAD"))
+		if err != nil {
+			log.Printf("error sending request: %s", err)
+			w.WriteHeader(500)
+			return
+		}
+		defer deadResp.Body.Close()
+
+		body, err = ioutil.ReadAll(deadResp.Body)
+		if err != nil {
+			log.Printf("error reading response body: %s", err)
+			w.WriteHeader(500)
+			return
+		}
+
+		cleanedBody = cleanBody(string(body))
+		log.Printf("received response %s from url %s", cleanedBody, os.Getenv("DEAD"))
+		if cleanedBody != deadServiceBody {
+			log.Printf("unexpected response body: %s expected: %s", cleanedBody, deadServiceBody)
+			w.WriteHeader(500)
+			return
+		}
+
+		// not found tests - 404 error codes
+		notFoundResp, err := client.Get(os.Getenv("NOT_FOUND"))
+		if err != nil {
+			log.Printf("error sending request: %s", err)
+			w.WriteHeader(500)
+			return
+		}
+		defer notFoundResp.Body.Close()
+
+		body, err = ioutil.ReadAll(notFoundResp.Body)
+		if err != nil {
+			log.Printf("error reading response body: %s", err)
+			w.WriteHeader(500)
+			return
+		}
+
+		cleanedBody = cleanBody(string(body))
+		log.Printf("received response %s from url %s", cleanedBody, os.Getenv("TEST_URL"))
+		if cleanedBody != notFoundBody {
+			log.Printf("unexpected response body: %s expected: %s", cleanedBody, notFoundBody)
+			w.WriteHeader(500)
+			return
```

Comment:
i may be missing something but if all checks are valid you'd still want to return a 200 right?

Reply:
I was following the same pattern from the other embedded clients where the only thing that needed to be returned was a 500 in case of errors. I can ask Oliver about the reason there aren't 200s returning but it does work